Libplanet/Store/Trie/MerkleTrie.cs

```diff
@@ -102,42 +124,60 @@ public ITrie Commit(bool rehearsal = false)
         }
 
         internal IEnumerable<HashDigest<SHA256>> IterateHashNodes()
+        {
+            return IterateNodes().Where(pair => pair.Node is HashNode)
+                .Select(pair => ((HashNode)pair.Node).HashDigest);
+        }
+
+        internal IEnumerable<(INode Node, ImmutableArray<byte> Path)> IterateNodes()
         {
             if (Root is null)
             {
                 yield break;
             }
 
-            var queue = new Queue<HashNode>();
-            if (!(Root is HashNode))
-            {
-                yield return Root.Hash();
-            }
-            else
-            {
-                queue.Enqueue((Root as HashNode)!);
-            }
+            var queue = new Queue<(INode, ImmutableArray<byte>)>();
+            queue.Enqueue((Root, ImmutableArray<byte>.Empty));
 
             while (queue.Count > 0)
             {
-                HashNode hashNode = queue.Dequeue();
-                yield return hashNode.HashDigest;
-                INode? node = GetNode(hashNode.HashDigest);
+                (INode node, ImmutableArray<byte> path) = queue.Dequeue();
+                yield return (node, path);
                 switch (node)
                 {
                     case FullNode fullNode:
                         IEnumerable<HashNode> childHashNodes = fullNode.Children.OfType<HashNode>();
```

Comment:
This doesn't seem to be used.

Reply:
I amended in 5fdd86b432c7f8e3f18e3fbe5aefd663bb4ab59e. 🙏